integration-tests/utils/integration-helpers.js

```diff
@@ -122,18 +131,52 @@ const deleteWorkspace = withSignedInPage(async ({ page, billingProject, workspac
   }
 });
 
-const withWorkspace = (test) => async (options) => {
-  console.log('withWorkspace ...');
-  const workspaceName = await makeWorkspace(options);
+const deleteWorkspaceV2 = withSignedInPage(async ({ page, billingProject, workspaceName }) => {
+  try {
+    await page.evaluate(
+      async (name, billingProject) => {
+        await window.Ajax().Workspaces.workspaceV2(billingProject, name).delete();
+      },
+      workspaceName,
+      billingProject
+    );
+    console.info(`Deleted workspace: ${workspaceName}`);
+  } catch (e) {
+    console.error(`Failed to delete workspace: ${workspaceName} with billing project: ${billingProject}`);
+    console.error(e);
+    throw e;
+  }
+});
+
+const withGcpWorkspace = (test) => async (options) => {
+  console.log('withGcpWorkspace ...');
+  const { workspaceName } = await makeGcpWorkspace(options);
 
   try {
     await test({ ...options, workspaceName });
   } finally {
-    console.log('withWorkspace cleanup ...');
+    console.log('withGcpWorkspace cleanup ...');
     await deleteWorkspace({ ...options, workspaceName });
   }
 };
 
+const withAzureWorkspace = (test) => async (options) => {
+  console.log('withAzureWorkspace ...');
+  options = {
+    ...options,
+    billingProject: options.billingProjectAzure,
+  };
+  const { workspaceName } = await makeWorkspace(options);
+  console.log(`withAzureWorkspace made workspace ${workspaceName}`);
+
+  try {
+    await test({ ...options, workspaceName });
+  } finally {
+    console.log('withAzureWorkspace cleanup ...', options, workspaceName);
+    await deleteWorkspaceV2({ ...options, workspaceName });
```

Comment:
Do you know that the workspace is in a deletable state when this is called? Also be aware that this method is async, so it will return quickly even though the actual workspace deletion will take a long time. That shouldn't be a big deal  when running the test in a long-lived environment like dev and staging, except that if the deletion ultimately fails, resources will build up.

There is a janitor "test" that attempts to delete old orphaned workspaces by deleting any workspaces with the prefix `testWorkspaceNamePrefix` that are at least 30 days old. I don't know if this will have any luck with Azure workspaces that fail to delete the first time (assuming you are using the prefix).

Reply:
The only Azure workspaces I've had fail to delete were ones created by tests which bombed out early enough to attempt this call when a WDS app were still in Provisioning status. Those, I doubt the janitor will be able to clean up. On the whole, since this isn't intended to be a test of workspace deletion, I'm comfortable with the way this is working now. What do you think?

Reply:
I'm worried that resources will end up leaking and build up. But we should wait until the discussion has been had about when/how often these tests run, and if you are going to switch them to run on a fixture workspace (in which case you wouldn't want to delete the workspace).

Reply:
Agreed!

Reply:
Yeah... so this is definitely something to think about.

If we are using a static workspace, and dont call delete, the risk of resources building up is fairly low since leo restricts a user to one runtime per workspace.

I do see a possible failure case where the test bombs out, doesn't make it to delete, and then all subsequent runs of the tests will fail being an issue 🤔 

(source: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala#L108)